sources/packages/backend/apps/api/src/route-controllers/assessment/_tests_/e2e/assessment.students.controller.getAssessmentAwardDetails.e2e-spec.ts

```diff
@@ -0,0 +1,591 @@
+import { HttpStatus, INestApplication } from "@nestjs/common";
+import * as request from "supertest";
+import {
+  BEARER_AUTH_TYPE,
+  createTestingAppModule,
+  FakeStudentUsersTypes,
+  getStudentToken,
+  mockUserLoginInfo,
+} from "../../../../testHelpers";
+import {
+  saveFakeApplicationDisbursements,
+  E2EDataSources,
+  createE2EDataSources,
+  saveFakeStudent,
+  createFakeDisbursementValue,
+  MSFAAStates,
+  createFakeMSFAANumber,
+  saveFakeDisbursementReceiptsFromDisbursementSchedule,
+} from "@sims/test-utils";
+import {
+  ApplicationStatus,
+  COEStatus,
+  DisbursementValueType,
+  MSFAANumber,
+  OfferingIntensity,
+  Student,
+} from "@sims/sims-db";
+import { getDateOnlyFormat } from "@sims/utilities";
+import { BC_TOTAL_GRANT_AWARD_CODE } from "@sims/services/constants";
+
+describe("AssessmentStudentsController(e2e)-getAssessmentAwardDetails", () => {
+  let app: INestApplication;
+  let db: E2EDataSources;
+  let sharedStudent: Student;
+  let sharedMSFAANumber: MSFAANumber;
+
+  beforeAll(async () => {
+    const { nestApplication, module, dataSource } =
+      await createTestingAppModule();
+    app = nestApplication;
+    db = createE2EDataSources(dataSource);
+    // Shared student to be used for tests that would not need further student customization.
+    sharedStudent = await saveFakeStudent(db.dataSource);
+    mockUserLoginInfo(module, sharedStudent);
+    // Valid MSFAA Number that will be part of the expected returned values.
+    sharedMSFAANumber = await db.msfaaNumber.save(
+      createFakeMSFAANumber(
+        { student: sharedStudent },
+        { msfaaState: MSFAAStates.Signed },
+      ),
+    );
+  });
+
+  it("Should get the student assessment summary containing loan and all grants for a part-time application with a single disbursement.", async () => {
+    // First disbursement values.
+    const firstDisbursementValues = [
+      createFakeDisbursementValue(
+        DisbursementValueType.CanadaLoan,
+        "CSLP",
+        111,
+      ),
+      createFakeDisbursementValue(
+        DisbursementValueType.CanadaGrant,
+        "CSGP",
+        222,
+      ),
+      createFakeDisbursementValue(
+        DisbursementValueType.CanadaGrant,
+        "CSPT",
+        333,
+      ),
+      createFakeDisbursementValue(
+        DisbursementValueType.CanadaGrant,
+        "CSGD",
+        444,
+      ),
+      createFakeDisbursementValue(DisbursementValueType.BCGrant, "BCAG", 555),
+      createFakeDisbursementValue(DisbursementValueType.BCGrant, "SBSD", 666),
+      createFakeDisbursementValue(
+        DisbursementValueType.BCTotalGrant,
+        "BCSG",
+        555 + 666,
```

Comment:
Suggestion. probably we can use the whole value.

Reply:
The values expressed as an addition would be easier to understand (IMO) and see the correlation with the above values. We can be even more explicit with constants but I did not see a point in going that far.